clustering/web/extension/src/main/java/org/wildfly/extension/clustering/web/InfinispanSessionManagementServiceConfigurator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.clustering.web;
+
+import static org.wildfly.extension.clustering.web.InfinispanSessionManagementResourceDefinition.Attribute.CACHE;
+import static org.wildfly.extension.clustering.web.InfinispanSessionManagementResourceDefinition.Attribute.CACHE_CONTAINER;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.clustering.service.ServiceConfigurator;
+import org.wildfly.clustering.web.infinispan.session.InfinispanSessionManagementConfiguration;
+import org.wildfly.clustering.web.infinispan.session.InfinispanSessionManagementProvider;
+import org.wildfly.clustering.web.session.DistributableSessionManagementProvider;
+
+/**
+ * Service configurator for Infinispan session management providers.
+ * @author Paul Ferraro
+ */
+public class InfinispanSessionManagementServiceConfigurator extends SessionManagementServiceConfigurator implements InfinispanSessionManagementConfiguration {
+
+    private volatile String containerName;
+    private volatile String cacheName;
```

Comment:
These fields are volatile since they are read by a different thread than the thread that sets them.  The getters in question are invoked after the configure(...) method is called (i.e. when then service that exposes the interface containing those getters starts), so there is no potential race condition.